src/containers/my-quizzes/create-or-edit-quiz-question/CreateOrEditQuizQuestion.tsx

```diff
@@ -1,108 +1,76 @@
+import { useImperativeHandle, forwardRef, useCallback } from 'react'
 import Box from '@mui/material/Box'
 
 import { useModalContext } from '~/context/modal-context'
+import useSnackbarAlert from '~/hooks/use-snackbar-alert'
 import { ResourceService } from '~/services/resource-service'
 import useForm from '~/hooks/use-form'
-import useAxios from '~/hooks/use-axios'
+import useMutation from '~/hooks/use-mutation'
 import QuestionEditor from '~/components/question-editor/QuestionEditor'
 import CreateOrEditQuestionModal from '~/containers/my-resources/create-or-edit-question-modal/CreateOrEditQuestionModal'
 
-import { getErrorMessage } from '~/utils/error-with-message'
 import { snackbarVariants } from '~/constants'
 import {
   ComponentEnum,
-  ErrorResponse,
-  Question,
-  QuestionForm,
-  QuestionModalForm,
-  UpdateQuestionParams
+  type Question,
+  type QuestionForm,
+  type QuestionModalForm
 } from '~/types'
 import { initialValues } from '~/containers/my-quizzes/create-or-edit-quiz-question/CreateOrEditQuizQuestion.constants'
-import { useAppDispatch } from '~/hooks/use-redux'
-import { openAlert } from '~/redux/features/snackbarSlice'
-import { getErrorKey } from '~/utils/get-error-key'
-import { useCallback, useEffect, useState } from 'react'
 
 interface CreateOrEditQuizQuestionProps {
   question?: Question
   setQuestions: React.Dispatch<React.SetStateAction<Question[]>>
   onCancel: () => void
 }
 
-const CreateOrEditQuizQuestion: React.FC<CreateOrEditQuizQuestionProps> = ({
-  question,
-  setQuestions,
-  onCancel
-}) => {
-  const dispatch = useAppDispatch()
-  const [isNewQuestion, setIsNewQuestion] = useState<boolean>(!!question)
-  const { openModal, closeModal } = useModalContext()
-
-  const createQuestionService = useCallback(
-    (data?: QuestionForm) => ResourceService.createQuestion(data),
-    []
-  )
+export interface CreateOrEditQuizQuestionRef {
+  openCreateModal: () => void
+}
 
-  const updateQuestionService = useCallback(
-    (params?: UpdateQuestionParams) => ResourceService.updateQuestion(params),
-    []
-  )
+const CreateOrEditQuizQuestionComponent = (
+  { question, setQuestions, onCancel }: CreateOrEditQuizQuestionProps,
+  ref: React.Ref<CreateOrEditQuizQuestionRef>
+) => {
```

Comment:
Please, always use `forwardRef` for components that can handle references

```suggestion
const CreateOrEditQuizQuestionComponent = forwardRef<{ openCreateModal: () => void }, CreateOrEditQuizQuestionProps>(
  { question, setQuestions, onCancel }, reference
) => {
```